Sources/Basic/Sequence+Duplicates.swift

```diff
@@ -0,0 +1,33 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+public extension Sequence where Iterator.Element: Hashable {
+    // Returns the set of duplicate elements in two arrays, if any.
+    func duplicates(_ other: [Iterator.Element]) -> Set<Iterator.Element>? {
```

Comment:
I am not sure that this API is good enough for Basic. Lets keep it in `PackageGraphLoader.swift` and resolve it as part of SR-4859.